### PR TITLE
Snapshot release

### LINKED ISSUE #240: SNAPSHOT releases from the master
Would be nice to release snapshots from the **master** branch on every merge, so using the latest unreleased changes on other projects would be more convenient.

### PR DESCRIPTION
**Pull request by psycho-ir**
closes #240

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: Release to Maven Central
 on:
   release:
-    types: [created]
+    types: [ released ]
```
Comment (psycho-ir) [inline .github/workflows/release.yml:4]: Releasing on `created` event may be a little bit soon, Doing so on release guarantees that the build will take place just after the release is published.

**Comment by csviri** [comment]
@psycho-ir We already have a ci.yaml, where we run the tests etc in case of the branch push. I think:

- On push to master we should run the tests as in this CI and then release the snapshot. Currently I don't see the tests running   on this new pipeline.  
- On the other hand we should then remove the event on the CI.yaml pipeline, so it just runs on a PR. (Maybe also rename this to PR.yaml or such)

**Comment by psycho-ir** [comment]
> @psycho-ir We already have a ci.yaml, where we run the tests etc in case of the branch push. I think:
> 
> * On push to master we should run the tests as in this CI and then release the snapshot. Currently I don't see the tests running   on this new pipeline.
> * On the other hand we should then remove the event on the CI.yaml pipeline, so it just runs on a PR. (Maybe also rename this to PR.yaml or such)

Makes sense, will update it accordingly.

**Review by csviri — APPROVED** [review]
LGTM

**Review by kirek007 — APPROVED** [review]
Makes sense.Thanks for work! Everything looks good to me.

**Pull request merged**